src/traces/bar/calc.js

```diff
@@ -31,9 +33,25 @@ module.exports = function calc(gd, trace) {
     var serieslen = Math.min(pos.length, size.length);
     var cd = new Array(serieslen);
 
-    // set position and size
-    for(var i = 0; i < serieslen; i++) {
-        cd[i] = { p: pos[i], s: size[i] };
+    // set position and size (as well as for waterfall total size)
+    var previousSum = 0;
+    for(i = 0; i < serieslen; i++) {
+        cd[i] = {
+            p: pos[i],
+            s: size[i]
+        };
+
+        if(isWaterfall) {
+            if(cd[i].s === undefined) {
```

Comment:
So, in order to draw partial sums, one has to set `size` as an array with `undefined` entries corresponding to the location of the partial sums?

As I said before in https://github.com/plotly/plotly.js/issues/3037#issuecomment-454864262, I'm very much against that. But oh well, for a quick prototype, this is :ok_hand: